dynamo/preprocessing/preprocessor_utils.py

```diff
@@ -918,6 +484,65 @@ def log1p_inplace(adata: AnnData, layer: str = DKM.X_LAYER) -> None:
         _log1p_inplace(mat)
 
 
+def log2_inplace(adata: AnnData, layer: str = DKM.X_LAYER) -> None:
+    """Calculate log1p (log(1+x)) for a layer of an AnnData object inplace.
+
+    Args:
+        adata: an AnnData object.
+        layer: the layer to operate on. Defaults to DKM.X_LAYER.
+    """
+
+    mat = DKM.select_layer_data(adata, layer, copy=False)
+    if issparse(mat):
+        if is_integer_arr(mat.data):
+            mat = mat.asfptype()
+            DKM.set_layer_data(adata, layer, mat)
+        _log2_inplace(mat.data + 1)
+    else:
+        mat = mat.astype(np.float)
+        _log2_inplace(mat + 1)
```

Comment:
There might be a bug in the inplace function. The result may not be saved correctly because we add a scalar to the `mat` or `mat.data`. Same for log_inplace.

Reply:
Thank you for taking the time to thoroughly review the code. I have addressed the issue and made the necessary changes.